tests/core/data_layer/test_data_store.py

```diff
@@ -195,15 +203,14 @@ async def test_insert_internal_node_does_nothing_if_matching(data_store: DataSto
     ancestors = await data_store.get_ancestors(node_hash=kv_node.hash, tree_id=tree_id)
     parent = ancestors[0]
 
-    async with data_store.db_wrapper.locked_transaction():
-        cursor = await data_store.db.execute("SELECT * FROM node")
+    async with data_store.db_wrapper.reader() as reader:
+        cursor = await reader.execute("SELECT * FROM node")
```

Comment:
probably not for this PR, but there is an `execute_fetchall()` function which would save you one round-trip to the sqlite thread